src/Hangfire.Realm/RealmWriteOnlyTransaction.cs

```diff
@@ -207,6 +206,7 @@ public override void AddToQueue(string queue, string jobId)
         {
             var realm = _storage.GetRealm();
             realm.Write(() => realm.Add(new JobQueueDto { Queue = queue, JobId = jobId }));
+            _afterCommitCommandQueue.Enqueue(() => RealmJobQueue.NewItemInQueueEvent.Set());
```

Comment:
This looks lifted from the SQL implementation. If you take a look at that, you'll see that the `Queue<Action>` is being used to queue up the `NewItemInQueueEvent` because the SQL version isn't committing the transaction until later. You don't need to do that here (and in fact never actually do dequeue and execute this event) since you are immediately writing to Realm.

This could be simplified to

`RealmJobQueue.NewItemInQueueEvent.Set();`

And the private readonly Queue<Action> class property can be removed.

Reply:
Good catch